lib/bundler/rubygems_ext.rb

```diff
@@ -23,6 +23,21 @@
 module Gem
   @loaded_stacks = Hash.new {|h, k| h[k] = [] }
 
+  class StubSpecification
```

Comment:
Our spec doesn't call `extension_dir` at all since stub is what the `extension_dir` is called on. So `extension_dir` needs to be defined on `Gem::StubSpecification`